oauthproxy.go

```diff
@@ -609,6 +613,25 @@ func (p *OAuthProxy) IsWhitelistedPath(path string) bool {
 	return false
 }
 
+func (p *OAuthProxy) IsWhitelistedIP(req *http.Request) bool {
+	if p.bypassIPWhitelist == nil {
+		return false
+	}
+
+	remoteAddr, err := getRealIP(req)
+	if err != nil {
+		logger.Printf("Error obtaining real IP for whitelist: %s", err.Error())
+		// Possibly spoofed X-Real-IP header
+		return false
+	}
+
+	if remoteAddr == nil {
+		return false
+	}
+
+	return p.bypassIPWhitelist.has(*remoteAddr)
```

Comment:
IPRange? May be there is a golang lib which do that automatically

Reply:
They have `net.IPNet.Contains()`, but it won't scale to thousands of whitelist entries, because we'd have to iterate over every one.

There's [yl2chen/cidranger](https://github.com/yl2chen/cidranger), but I'm not sure whether or not they maintainers will love adding another dependancy.  Dependancies are a real pain in Golang, no-one cares to maintain backwards compatibility for their libraries and the built in package/library manager (`go get`) doesn't support any sort of fixed versioning.

Reply:
IMHO, go for net.IPNet.Contains()

Reply:
I hear you.  For us, it'd work fine.  But if we care about getting it upstreamed we need something that doesn't degrade performance too badly.